### PR TITLE
GH-1155 Table Sort Publish Param

### PR DESCRIPTION
**Pull request by dtsnell4**
Adds sortByFieldIndex and minWidgetHeight publish parameters to make the table sorting persistible.

Also adds the Utility.multiSort() as the sorting function.

Fixes GH-1155

Signed-off-by: Dan Snell <Dan.Snell@lexisnexis.com>

### CODE REVIEW

**Comment by dtsnell4** [comment]
@GordonSmith @mlzummo @jbrundage Please review

**Comment by dtsnell4** [comment]
http://rawgit.com/dtsnell4/Visualization/GH-1155_Table_Sort_Persist/demos/dermatology.html?src/other/Table&sortByFieldIndex=0&currentSortOrder=true

File: src/other/Table.js [hunk 1]
```diff
@@ -70,14 +68,8 @@
...
-    Table.prototype.data = function (_) {
-        var retVal = HTMLWidget.prototype.data.apply(this, arguments);
-        if (arguments.length) {
-            this._currentSort = "";
-            this._currentSortOrder = 1;
-        }
-        return retVal;
-    };
+    Table.prototype.publish("sortByFieldIndex", null, "number", "Index for the field/column to sort the data", null, { tags: ["Basic"], optional: true });
+    Table.prototype.publish("currentSortOrder", true, "boolean", "Direction for sorting the data: ascending (true) or decsending (false)", null, { tags: ["Basic"], optional: true });
```
Comment (GordonSmith) [inline src/other/Table.js]: Rename to "decsending" and default to false.

File: src/marshaller/HipieDDL.js [hunk 2]
```diff
@@ -407,6 +407,7 @@
         var db = this.getOutput().db;
         var retVal = this.mappings.doMapAll(db);
         if (retVal.length && this.sort) {
+            console.log(retVal, db.hipieMapSortArray(this.sort));
```
Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: Remove console log.

**Comment by GordonSmith** [comment]
@dtsnell4 - the first time a columns is clicked to be sorted, it sorts descending, it should be ascending.

**Comment by dtsnell4** [comment]
@GordonSmith @mlzummo @jbrundage  Updated 

http://rawgit.com/dtsnell4/Visualization/GH-1155_Table_Sort_Persist/demos/dermatology.html?src/other/Table&sortByFieldIndex=0&descending=true

**Comment by jbrundage** [comment]
Looks good to me

**Comment by mlzummo** [comment]
looks good to me too

**Pull request merged**